internal/rancher/client.go

```diff
@@ -29,20 +29,22 @@ func NewClient() *Client {
 }
 
 type Client struct {
-	c           *http.Client
-	cachedToken KubeconfigToken
-	mux         sync.Mutex
-	password    string
-	timeout     time.Duration
-	url         string
-	username    string
+	c               *http.Client
+	cachedToken     KubeconfigToken
+	mux             sync.Mutex
+	password        string
+	shortExpiration int // seconds for the expiration
+	timeout         time.Duration
+	url             string
+	username        string
 }
 
 func (c *Client) Token(ctx context.Context) (string, error) {
 	c.mux.Lock()
 	defer c.mux.Unlock()
 
-	if time.Now().In(time.UTC).After(c.cachedToken.ExpiresAt) || c.cachedToken.Token == "" {
+	tokenExpired := int(time.Since(c.cachedToken.Created.In(time.UTC)).Seconds()) > c.shortExpiration
```

Comment:
What if short expiration is less than or equal to 0?

Reply:
Well, before we weren't setting it at all if it wan't > 0 but I moved the >0 logic to the `tokenExpired` method which addresses this (also fixes the failing test :) - that's b/c since it defaulted to 0 it was always hitting this even though the shortExpiration wasn't "set").